File: dev-docs/snippets/groups/create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a new group. `groupType`, `groupId`, and `name` are required.
export async function create(client: JsBaoClient) {
  // #region example
  const group = await client.groups.create({
    groupType: "team",
    groupId: "design",
    name: "Design Team",
    description: "Product design crew",
  });
  // #endregion example
  return group;
}

File: dev-docs/snippets/collections/create.swift
import JsBaoClient

// Create a new collection. `name` is required; `collectionType`/`contextId`
// are immutable after create.
func create(client: JsBaoClient) async throws {
  // #region example
  let collection = try await client.collections.create(
    params: CreateCollectionParams(
      name: "Q3 Planning",
      description: "Docs for the Q3 planning cycle",
      collectionType: "default"
    )
  )
  // #endregion example
  _ = collection
}
